lib/prepare-write.js

```diff
@@ -44,7 +44,9 @@ function prepareWrite(outFolder, file, opt, callback) {
   file.stat.mode = options.mode;
   file.flag = options.flag;
   file.cwd = cwd;
-  file.base = basePath;
+  // Ensure the base always ends with a separator
+  // TODO: add a test for this
+  file.base = path.normalize(basePath + path.sep);
```

Comment:
I actually think this needs to be pushed down into vinyl but I will do that separately from this PR